pkg/api/v1/types_swagger_doc_generated.go

```diff
@@ -294,7 +295,7 @@ func (EndpointSubset) SwaggerDoc() map[string]string {
 var map_Endpoints = map[string]string{
 	"":         "Endpoints is a collection of endpoints that implement the actual service. Example:\n  Name: \"mysvc\",\n  Subsets: [\n    {\n      Addresses: [{\"ip\": \"10.10.1.1\"}, {\"ip\": \"10.10.2.2\"}],\n      Ports: [{\"name\": \"a\", \"port\": 8675}, {\"name\": \"b\", \"port\": 309}]\n    },\n    {\n      Addresses: [{\"ip\": \"10.10.3.3\"}],\n      Ports: [{\"name\": \"a\", \"port\": 93}, {\"name\": \"b\", \"port\": 76}]\n    },\n ]",
 	"metadata": "Standard object's metadata. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata",
-	"subsets":  "The set of all endpoints is the union of all subsets. Sets of addresses and ports that comprise a service.",
+	"subsets":  "The set of all endpoints is the union of all subsets. Addresses are placed into subsets according to the IPs they share. A single address with multiple ports, some of which are ready and some of which are not (because they come from different containers) will result in the address being displayed in different subsets for the different ports. No address will appear in both Addresses and NotReadyAddresses in the same subset. Sets of addresses and ports that comprise a service.",
```

Comment:
An address+port is ready or not.  An address that appears in one subset
will not appear in any other subset that has the same ports as the subset
the address is in.  An address can show up in two different subsets via
different ports, and have different readiness depending on whether that
particular port is ready.

On Wed, Sep 16, 2015 at 5:39 PM, Abhi Shah <notifications@github.com> wrote:

> In pkg/api/v1/types_swagger_doc_generated.go
> <https://github.com/kubernetes/kubernetes/pull/13778#discussion_r39688882>
> :
>
> > @@ -294,7 +295,7 @@ func (EndpointSubset) SwaggerDoc() map[string]string {
> >  var map_Endpoints = map[string]string{
> >  	"":         "Endpoints is a collection of endpoints that implement the actual service. Example:\n  Name: \"mysvc\",\n  Subsets: [\n    {\n      Addresses: [{\"ip\": \"10.10.1.1\"}, {\"ip\": \"10.10.2.2\"}],\n      Ports: [{\"name\": \"a\", \"port\": 8675}, {\"name\": \"b\", \"port\": 309}]\n    },\n    {\n      Addresses: [{\"ip\": \"10.10.3.3\"}],\n      Ports: [{\"name\": \"a\", \"port\": 93}, {\"name\": \"b\", \"port\": 76}]\n    },\n ]",
> >  	"metadata": "Standard object's metadata. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata",
> > -	"subsets":  "The set of all endpoints is the union of all subsets. Sets of addresses and ports that comprise a service.",
> > +	"subsets":  "The set of all endpoints is the union of all subsets. Addresses are placed into subsets according to the IPs they share. A single address with multiple ports, some of which are ready and some of which are not (because they come from different containers) will result in the address being displayed in different subsets for the different ports. No address will appear in both Addresses and NotReadyAddresses in the same subset. Sets of addresses and ports that comprise a service.",
>
> my understanding from your code is that, an address is either ready or
> not. We don't split the readiness of an address by port.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13778/files#r39688882>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift